ovirt/suite_test.go

```diff
@@ -0,0 +1,773 @@
+	timeout, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+
+	for {
+		_, err = o.conn.SystemService().NetworksService().NetworkService(network.MustId()).Remove().Send()
+		if err == nil {
+			return
+		}
+		select {
+		case <-timeout.Done():
+			return fmt.Errorf("timeout (%w)", err)
+		case <- time.After(10 * time.Second):
+		}
+	}
+}
+
+var letters = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+func (o *ovirtTestSuite) generateRandomID(n uint) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letters[o.rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+func (o *ovirtTestSuite) CreateDisk() (disk *ovirtsdk4.Disk, err error) {
+	diskName := fmt.Sprintf("test_disk_%s", o.generateRandomID(5))
+	disk, err = ovirtsdk4.NewDiskBuilder().
+		Name(diskName).
+		Format(ovirtsdk4.DISKFORMAT_RAW).
+		ProvisionedSize(1024*1024).
+		StorageDomainsOfAny(
+			ovirtsdk4.NewStorageDomainBuilder().
+				Id(o.storageDomainID).
+				MustBuild(),
+		).
+		Build()
+	if err != nil {
+		// This should never happen
+		panic(err)
+	}
+	diskResponse, err := o.conn.SystemService().DisksService().Add().Disk(disk).Send()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create disk (%w)", err)
+	}
+	disk, ok := diskResponse.Disk()
+	if !ok {
+		return nil, fmt.Errorf("no disk object returned from disk creation")
+	}
+
+	timeout, cancelFunc := context.WithTimeout(context.Background(), 5 * time.Minute)
+	defer cancelFunc()
+	for {
+		diskResponse, err := o.conn.SystemService().DisksService().DiskService(disk.MustId()).Get().Send()
+		if err == nil {
+			if disk, ok := diskResponse.Disk(); ok {
+				if diskStatus, ok := disk.Status(); ok {
+					if diskStatus == ovirtsdk4.DISKSTATUS_OK {
+						return disk, nil
+					}
+				}
+			}
+		}
+		select {
+		case <-timeout.Done():
+			return disk, fmt.Errorf("timeout while waiting for disk to come up")
+		case <- time.After(10 * time.Second):
+		}
+	}
+}
+
+func (o *ovirtTestSuite) RemoveDisk(disk *ovirtsdk4.Disk) (err error) {
+	_, err = o.conn.SystemService().DisksService().DiskService(disk.MustId()).Remove().Send()
+	if err != nil {
+		return fmt.Errorf("failed to remove disk (%w)", err)
+	}
+	return nil
+}
+
+func (o *ovirtTestSuite) GetTestDatacenterName() string {
+	return o.testDatacenterName
+}
+
+func (o *ovirtTestSuite) GetTestAuthzName() string {
+	return o.testAuthz
+}
+
+func (o *ovirtTestSuite) TestClusterDestroy(cluster *ovirtsdk4.Cluster) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		listResponse, err := o.conn.SystemService().ClustersService().List().Query("id", cluster.MustId()).Send()
+		if err != nil {
+			return err
+		}
+		if len(listResponse.MustClusters().Slice()) != 0 {
+			return fmt.Errorf("cluster %s has not been removed after test", cluster.MustName())
+		}
+		return nil
+	}
+}
+
+func (o *ovirtTestSuite) TestResourceAttrNotEqual(
```

Comment:
looks like a very general-purpose method, do we want it inside the suite?

Reply:
It has been moved from an old method that was there. I generally moved everything into the suite to keep things in one place.

Reply:
okay .. those kinds of methods are usually handled by testing frameworks, are we planning to integrate one here?

Reply:
No. The suite is the testing framework.